server/lib/chat/chat-models.ts

```diff
@@ -116,6 +117,41 @@ export async function getUserChannelEntriesForUser(
   }
 }
 
+export async function countUserJoinedChannels(
```

Comment:
It's bad to have these be separate queries, there's nothing stopping this from being changed in the middle of queries running. With stuff like this you either:

1) Need to lock a row that all other queries modifying this data will also lock (this requires transactions)
2) Ensure the limit in the same query (*query*, not transaction) that you're doing the update (e.g. have a where clause that won't apply if they're over the limit)

I think 2 likely makes more sense here, as we don't really have a good single row to lock and requiring transactions kinda sucks.